### PR TITLE
feat: insight for lcp

### PR DESCRIPTION
**Pull request by DomonJi**
FEATURE:
image
image

https://github.com/perfsee/perfsee/assets/9986655/a499159a-8249-4c70-8634-859772f3f60d

TODO:
- [x] source analysis for lcp

### CODE REVIEW

File: packages/flamechart/src/components/flamechart-view-container.tsx [hunk 1]
```diff
@@ -93,15 +95,16 @@ export const FlamechartViewContainer = memo(
         style,
         renderTooltip,
         renderTimingTooltip,
+        onClickTiming,
+        selectedFrame,
       }: FlamechartViewProps,
       ref: ForwardedRef<FlamechartViewContainerRef>,
     ) => {
-      const [selectedFrame, setSelectedFrame] = useState<FlamechartFrame | null>(null)
```
Comment (DomonJi) [inline packages/flamechart/src/components/flamechart-view-container.tsx:99]: @EYHN I put this state outside to make the selectedFrame controllable, which may have breaking changes to other users.

File: packages/flamechart/src/components/flamechart-group-container.tsx [hunk 2]
```diff
@@ -280,6 +288,9 @@ export const FlamechartGroupContainer = withErrorBoundary<React.FunctionComponen
               onSelectFrame={handleSelectFlamechart}
               renderTooltip={profilesRenderTooltip?.[index]}
               renderTimingTooltip={profilesRenderTimingTooltip?.[index]}
+              onClickTiming={onClickTiming}
+              selectedFrame={selectedFrame}
+              disableDetailView={index !== profiles.length - 1}
```
Comment (EYHN) [inline packages/flamechart/src/components/flamechart-group-container.tsx:293]: users can hide the flamechart and the details view will also be hidden, so the detail view should be enabled in the last visible split.

You can calculate the last visible split
```
let lastVisibleSplit = -1
for (let i = splitCollapsed.length - 1; i >= 0; i--) {
  if (splitCollapsed[i]) {
    lastVisibleSplit = i
  } else {
    break
  }
}
lastVisibleSplit += 1
```

File: packages/flamechart/src/components/flamechart-group-container.tsx [hunk 3]
```diff
@@ -360,7 +371,6 @@ export const FlamechartGroupContainer = withErrorBoundary<React.FunctionComponen
                 {views}
               </SplitView>
             )}
-            {selectedFrame && <DetailView frame={selectedFrame} theme={globalTheme} />}
```
Comment (EYHN) [inline packages/flamechart/src/components/flamechart-group-container.tsx:363]: Make it an option to select this old simple detail view

File: packages/flamechart/src/lib/profile-search.ts [hunk 4]
```diff
@@ -2,23 +2,24 @@ import { Frame, CallTreeNode, Profile } from './profile'
...
+import { FlamechartFrame } from './flamechart'
 
 export enum FlamechartType {
   CHRONO_FLAME_CHART,
   LEFT_HEAVY_FLAME_GRAPH,
 }
 
 export interface ProfileSearchEngine {
-  getMatchForNode(node: CallTreeNode): FuzzyMatch | null
+  getMatchForFrame(frame: FlamechartFrame): FuzzyMatch | null
```
Comment (EYHN) [inline packages/flamechart/src/lib/profile-search.ts:13]: Why this change necessary?

**Review by EYHN — CHANGES_REQUESTED** [review]

File: packages/flamechart/src/lib/profile-search.ts (hunk 4 above)
Comment (DomonJi) [reply packages/flamechart/src/lib/profile-search.ts:13]: Used to match for frame's start time and endTime

**Review by EYHN — APPROVED** [review]

**Pull request merged**